kyaml/fn/framework/command/command_test.go

```diff
@@ -0,0 +1,170 @@
+	d, err := ioutil.TempDir("", "kustomize")
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	defer os.RemoveAll(d)
+
+	// create a function
+	cmd := command.Build(&framework.SimpleProcessor{}, false, false)
+
+	// generate the Dockerfile
+	cmd.SetArgs([]string{"gen", d})
+	if !assert.NoError(t, cmd.Execute()) {
+		t.FailNow()
+	}
+
+	b, err := ioutil.ReadFile(filepath.Join(d, "Dockerfile"))
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	expected := `FROM golang:1.15-alpine as builder
+ENV CGO_ENABLED=0
+WORKDIR /go/src/
+COPY . .
+RUN go build -v -o /usr/local/bin/function ./
+
+FROM alpine:latest
+COPY --from=builder /usr/local/bin/function /usr/local/bin/function
+CMD ["function"]
+`
+	if !assert.Equal(t, expected, string(b)) {
+		t.FailNow()
+	}
+}
+
+// TestCommand_standalone tests the framework works in standalone mode
+func TestCommand_standalone(t *testing.T) {
+	// TODO: make this test pass on windows -- currently failure seems spurious
+	testutil.SkipWindows(t)
+
+	type api struct {
+		A string `json:"a" yaml:"a"`
+	}
+	var config api
+
+	fn := func(items []*yaml.RNode) ([]*yaml.RNode, error) {
+		items = append(items, yaml.MustParse(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+ name: bar1
+ namespace: default
+ annotations:
+   foo: bar1
+`))
+		for i := range items {
+			err := items[i].PipeE(yaml.SetAnnotation("a", config.A))
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		return items, nil
+	}
+
+	cmdFn := func() *cobra.Command {
+		return command.Build(&framework.SimpleProcessor{Filter: kio.FilterFunc(fn), Config: &config}, false, false)
+	}
+
+	tc := frameworktestutil.CommandResultsChecker{Command: cmdFn}
+	tc.Assert(t)
+}
+
+func TestCommand_standalone_stdin(t *testing.T) {
+	// TODO: make this test pass on windows -- currently failure seems spurious
+	testutil.SkipWindows(t)
+
+	type api = struct {
+		A string `json:"a" yaml:"a"`
+	}
+	var config api
+
+	p := &framework.SimpleProcessor{
+		Config: &config,
+
+		Filter: kio.FilterFunc(func(items []*yaml.RNode) ([]*yaml.RNode, error) {
+			items = append(items, yaml.MustParse(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: bar2
+  namespace: default
+  annotations:
+    foo: bar2
+`))
+			for i := range items {
+				err := items[i].PipeE(yaml.SetAnnotation("a", config.A))
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			return items, nil
+		}),
+	}
+	cmd := command.Build(p, false, false)
+	cmd.SetIn(bytes.NewBufferString(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: bar1
+  namespace: default
+  annotations:
+    foo: bar1
+spec:
+  replicas: 1
+`))
+	var out bytes.Buffer
+	cmd.SetOut(&out)
+	cmd.SetArgs([]string{filepath.Join("testdata", "command", "config.yaml"), "-"})
+
+	require.NoError(t, cmd.Execute())
+
+	require.Equal(t, strings.TrimSpace(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: bar1
+  namespace: default
+  annotations:
+    foo: bar1
+    a: 'b'
```

Comment:
I'm missing where this `'b'` value is coming from and why it has the extra single quotes

Reply:
Set annotation uses single quotes -- Annotations should always be quoted to ensure they are correctly parsed as strings rather than other value types.  For instance if you set the value to `6`, the config would be invalid if it wasn't quoted.